pkg/stack_mgr/stack_service.go

```diff
@@ -137,21 +137,17 @@ func (s *StackService) Remove(ctx context.Context, stackName string) error {
 	if err != nil {
 		return errors.Wrap(err, "unable to get a list of stacks")
 	}
-	stackNames := map[string]bool{}
-	for currentStackName := range stacks {
-		if currentStackName != stackName {
-			stackNames[currentStackName] = true
-		}
-	}
 	stackNamesList := []string{}
-	for stackName := range stackNames {
-		stackNamesList = append(stackNamesList, stackName)
+	for name := range stacks {
+		if name != stackName {
+			stackNamesList = append(stackNamesList, name)
+		}
 	}
 
 	sort.Strings(stackNamesList)
 	stackNamesJson, err := json.Marshal(stackNamesList)
 	if err != nil {
-		return errors.Wrap(err, "unable to deserialize a stack list json")
+		return errors.Wrap(err, "unable to serialize stack list as json")
 	}
 	err = s.backend.WriteParam(ctx, s.writePath, string(stackNamesJson))
```

Comment:
not for this PR:
I think the order here is wrong. We should first delete from TFE and then delete from the stacklist. TFE delete should also probably wait. 

If we delete from stacklist and then TFE fails, we now have orphaned the stack.